projects/components/src/viewer/code-viewer/code-viewer.component.ts

```diff
@@ -211,17 +211,17 @@ export class CodeViewerComponent implements AfterViewInit, OnChanges, OnDestroy
     });
   }
 
-  private isLineHighlighted(lineNum: number): boolean {
+  private isLineHighlighted(codeText: string): boolean {
     if (typeof this.highlightText === 'string') {
-      return this.codeTexts[lineNum].toLowerCase().includes(this.highlightText.toLowerCase());
+      return codeText.includes(this.highlightText.toLowerCase());
     }
 
     if (Array.isArray(this.highlightText)) {
-      return this.highlightText.some(text => this.codeTexts[lineNum].toLowerCase().includes(text.toLowerCase()));
+      return this.highlightText.some(text => codeText.includes(text.toLowerCase()));
     }
 
     if (this.highlightText instanceof RegExp) {
-      return (this.codeTexts[lineNum].match(this.highlightText) ?? []).length > 0;
+      return (codeText.match(this.highlightText) ?? []).length > 0;
```

Comment:
qq.. You were not using lowercase here before. Would this change cause any issues?

Reply:
Nope

Reply:
Since regex will come from parent, it shouldn't matter